internal/service/appstream/user.go

```diff
@@ -0,0 +1,239 @@
+package appstream
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/appstream"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+)
+
+func ResourceUser() *schema.Resource {
+	return &schema.Resource{
+		CreateWithoutTimeout: resourceUserCreate,
+		ReadWithoutTimeout:   resourceUserRead,
+		UpdateWithoutTimeout: resourceUserUpdate,
+		DeleteWithoutTimeout: resourceUserDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"authentication_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice(appstream.AuthenticationType_Values(), false),
+			},
+			"created_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Computed: true,
+			},
+			"first_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(0, 2048),
+			},
+			"last_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(0, 2048),
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"user_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+		},
+	}
+}
+
+func resourceUserCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).AppStreamConn
+
+	userName := d.Get("user_name").(string)
+	authType := d.Get("authentication_type").(string)
+
+	input := &appstream.CreateUserInput{
+		AuthenticationType: aws.String(authType),
+		UserName:           aws.String(userName),
+	}
+
+	if v, ok := d.GetOk("first_name"); ok {
+		input.FirstName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("last_name"); ok {
+		input.LastName = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("message_action"); ok {
+		input.MessageAction = aws.String(v.(string))
+	}
```

Comment:
The `message_action` parameter should be restored. In https://github.com/hashicorp/terraform-provider-aws/pull/21485#discussion_r745155897, I was suggesting removing `RESEND` as a valid option and ignoring changes to the value instead of making it `ForceNew`. There should still be the options of sending the welcome email or setting `SUPPRESS` to not send the email.
Another option would be to use a `TypeBool` `send_welcome_email` (or `send_email_notification` to match `aws_appstream_user_stack_association`)

Reply:
Apologies, I misunderstood, thanks for the clarification.